File: Data Structures & Algorithms/duplicate-integer/submission-0.java

class Solution {
    public boolean hasDuplicate(int[] nums) {
        HashMap<Integer,Integer> map = new HashMap<>();
        int tmp = 0; 
        for (int key : nums) {
            tmp = 0; 
            if (map.containsKey(key)) {
                tmp = map.get(key);
                tmp++;
                map.put(key, tmp);
            }
            else {
                map.put(key, 1);
            }
        }

        for (Integer key : map.keySet()) {
            if (map.get(key) > 1) {
                return true;
            }
        }
        return false;
    }
}

File: Data Structures & Algorithms/majority-element/submission-0.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        map = {}
        for ele in nums: 
            if ele in map: 
                map[ele] += 1
            else: 
                map[ele] =1 
        
        for ele in map: 
            if map[ele] > (len(nums)/2):
                return ele
                
